cmake/FindMOSEK.cmake

```diff
@@ -0,0 +1,199 @@
+        SET(MOSEK_PLATFORM_NAME "linuxaarch64")
+    else()
+        SET(MOSEK_PLATFORM_NAME "linux64x86")
+    endif()
+elseif(MSVC)
+    SET(MOSEK_PLATFORM_NAME "win64x86")
+else()
+    message(FATAL_ERROR "Mosek not supported for ${CMAKE_SYSTEM}")
+endif()
+
+function(FindMosekPlatformInPath RESULT PATH)
+    SET(${RESULT} "")
+    if(EXISTS "${PATH}/mosek")
+        SET(dirlist "")
+        FILE(GLOB entries LIST_DIRECTORIES true "${PATH}/mosek/*")
+        FOREACH(f ${entries})
+            if(IS_DIRECTORY "${f}")
+                get_filename_component(bn "${f}" NAME)
+                if("${bn}" MATCHES "^[0-9]+[.][0-9]+$") 
+                    if (${bn} GREATER_EQUAL "10.0")
+                        LIST(APPEND dirlist "${bn}")
+                    endif()
+                endif()
+            endif()
+        ENDFOREACH()
+        LIST(SORT dirlist COMPARE NATURAL ORDER DESCENDING)
+
+        if(MOSEK_PLATFORM_NAME) 
+            foreach(MOSEK_VERSION ${dirlist})
+                SET(MSKPFDIR "${PATH}/mosek/${MOSEK_VERSION}/tools/platform/${MOSEK_PLATFORM_NAME}")
+                if(EXISTS "${MSKPFDIR}")
+                    SET(${RESULT} ${MSKPFDIR} PARENT_SCOPE)
+                    return()
+                endif()
+            endforeach()
+        endif()
+    endif()
+endfunction()
+
+
+function(MosekVersionFromPath RESVMAJOR RESVMINOR PATH)
+    execute_process(COMMAND "${PATH}/bin/mosek" "-v" RESULT_VARIABLE CMDRES OUTPUT_VARIABLE VERTEXT)
+    if (${CMDRES} EQUAL 0)
+        if (${VERTEXT} MATCHES "^MOSEK version [0-9]+[.][0-9]+[.][0-9]+")
+            string(REGEX REPLACE "^MOSEK version ([0-9]+)[.]([0-9]+).*" "\\1;\\2" MSKVER ${VERTEXT})
+
+            list(GET MSKVER 0 VMAJOR)
+            list(GET MSKVER 1 VMINOR)
+            set(${RESVMAJOR} "${VMAJOR}" PARENT_SCOPE)
+            set(${RESVMINOR} "${VMINOR}" PARENT_SCOPE)
+
+            #message(STATUS "mskver = ${VMAJOR}.${VMINOR} -> ${${RESVMAJOR}}.${${RESVMINOR}}")
+            #message(STATUS "    ${RESVMAJOR} = ${${RESVMAJOR}}")
+            #message(STATUS "    ${RESVMINOR} = ${${RESVMINOR}}")
+            return()
+        endif()
+    endif()
+endfunction()
+
+
+# Where to look for MOSEK:
+# Standard procedure in Linux/OSX is to install MOSEK in the home directory, i.e.
+#   $HOME/mosek/X.Y/...
+# Option 1. The user can specify when running CMake where the MOSEK platform directory is located, e.g.
+#     -DMOSEK_PLATFORM_DIR=$HOME/mosek/10.2/tools/platform/linux64x86/ 
+#   in which case no search is performed.
+# Option 2. The user can specify MOSEK_ROOT when running cmake. MOSEK_ROOT is
+#   the directory where the root mosek/ tree is located.
+# Option 3. Automatic search. We will then attempt to search in the default
+#   locations, and if that fails, assume it is installed in a system location.
+# For option 2 and 3, the newest MOSEK version will be chosen if more are available.
+
+if(MOSEK_PLATFORM_DIR)
```

Comment:
I would use `MOSEK_ROOT_DIR` to follow convention

From `CMake` head repository:
```sh
$ git grep -n "ROOT_DIR\b" | wc -l
209
$ git grep -n "PLATFORM_DIR\b" | wc -l
0
```